src/abstractdataframe/reshape.jl

```diff
@@ -185,39 +193,72 @@ long = stack(wide)
 wide0 = unstack(long)
 wide1 = unstack(long, :variable, :value)
 wide2 = unstack(long, :id, :variable, :value)
+wide3 = unstack(long, [:id, :a], :variable, :value)
 ```
 Note that there are some differences between the widened results above.
-
 """
 function unstack(df::AbstractDataFrame, rowkey::Int, colkey::Int, value::Int)
-    # `rowkey` integer indicating which column to place along rows
-    # `colkey` integer indicating which column to place along column headers
-    # `value` integer indicating which column has values
-    refkeycol = CategoricalArray{Union{eltype(df[rowkey]), Missing}}(df[rowkey])
-    # make sure we report only levels actually present in rowkey column
-    # this is consistent with how the other unstack method based on groupby works
+    refkeycol = copy(CategoricalArray{Union{eltype(df[rowkey]), Missing}}(df[rowkey]))
     droplevels!(refkeycol)
-    valuecol = df[value]
     keycol = CategoricalArray{Union{eltype(df[colkey]), Missing}}(df[colkey])
+    valuecol = df[value]
+    Nrow = length(refkeycol.pool)
+    Ncol = length(keycol.pool)
+    df2m = [similar_missing(valuecol, Nrow) for i in 1:Ncol]
```

Comment:
Why create this here? Don't you have all the needed information inside `_unstack`?

Reply:
bacause it is ~5% faster this way. I guess Julia does not like functions with large bodies and `_unstack` is already heavy. I do not know why in detail - those are benchmark results.

Reply:
5% doesn't sound worth it to me. And this kind of difference can change with compiler improvements. Better prioritize the clearer design.

Reply:
BTW, better not call it `df` since it's not a data frame.

Reply:
Moved and changed name to `unstacked_val`